### DISCUSSION TITLE
Automerging a group from config.js

### DISCUSSION THREAD

**Discussion by nickstolwijk** [renovatebot/renovate#8631] [Request Help] [answered]
I wanted to write some config for a renovate.json file that makes a group from the config.js automerge.

config.js:
```
packageRules: [
        {
            "groupName": "company parents",
            "matchPackagePatterns": [ "^nl.company:company*" ]
        },
    ]
```
renovate.js:
```
  "packageRules": [
    {
      "groupName": "company parents",
      "automerge": true
    }
  ]
```

This gave me a configuration error:
```
The renovate configuration file contains some invalid settings
packageRules: Each packageRule must contain at least one selector (matchPaths, matchLanguages, matchBaseBranches, matchManagers, matchDatasources, matchDepTypes, matchPackageNames, matchPackagePatterns, excludePackageNames, excludePackagePatterns, matchCurrentVersion, matchSourceUrlPrefixes, matchUpdateTypes). If you wish for configuration to apply to all packages, it is not necessary to place it inside a packageRule at all.",
```

Do I have the wrong mindset or is there a way to make this work? (Or should I just repeat the `matchPackagePatterns`?)

**Comment by viceice** [comment, marked as answer]
Instead of group name you need to repeat the matcher

**Reply by nickstolwijk** [reply]
But when the matcher changes in the future, I need to update all of my renovate.json files. Is there a good reason that I can not name a group there? (Or would it be a nice, new feature?)

**Reply by rarkins** [reply]
This is why you should avoid having these types of rules in you config.js and instead use config presets which your renovate.json extends. You could define a preset that purely does the matching, and then the other presets extend that.

**Reply by rarkins** [reply]
Backend rules in config.js are "invisible" to regular repository users so make their life harder. e.g. they'd have to figure out why there's even a "company parents" group existing in the first place when they haven't configured it. If you define a shared preset in a repo then they can trace back the config and extend/override it.

**Reply by nickstolwijk** [reply]
Thanks for the answer! Just started running Renovate on our repositories on Bitbucket Server.

The shared presets configuration is the next thing I will look at. :-)

**Reply by rarkins** [reply]
Example of a base pattern for extending: https://docs.renovatebot.com/presets-monorepo/#monorepoaws-java-sdk
Example of a grouping preset which extends it: https://docs.renovatebot.com/presets-group/#groupaws-java-sdkmonorepo

**Reply by nickstolwijk** [reply]
My first POC would be to have a shared preset in Bitbucket Server working for a repository. ;-)
